staking/programs/integrity-pool/src/utils/types.rs

```diff
@@ -5,3 +5,61 @@ pub type frac64 = u64;
 
 pub const FRAC_64_MULTIPLIER: u64 = 1_000_000;
 pub const FRAC_64_MULTIPLIER_U128: u128 = FRAC_64_MULTIPLIER as u128;
+
+
+pub struct BoolArray {
+    pub data: Vec<u8>,
+}
+
+impl BoolArray {
+    pub fn new(n: usize) -> Self {
+        Self {
+            data: vec![0; (n + 7) / 8],
+        }
+    }
+
+    pub fn get(&self, i: usize) -> bool {
+        let byte = i / 8;
+        let bit = i % 8;
+        self.data[byte] & (1 << bit) != 0
+    }
+
+    pub fn set(&mut self, i: usize, value: bool) {
```

Comment:
nitpick : we actually only use the `set(i, true)` so you could remove the case where it's false

Reply:
done